segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/mocks/InMemoryStorage.java

```diff
@@ -421,7 +421,9 @@ private void checkOpened() {
 
         @Override
         public String toString() {
-            return String.format("%s: Length = %d, Sealed = %s", this.name, this.length, this.sealed);
+            synchronized (this.lock) {
+                return String.format("%s: Length = %d, Sealed = %s", this.name, this.length, this.sealed);
```

Comment:
Is it sufficient if the variables read here are volatile?

Reply:
It would have been very sufficient. However my IDE doesn't know that `toString` is only used for debugging so every time I look at this file and compile it tells me I'm doing something bad.